### ISSUE TITLE
Interpolation syntax

### ISSUE THREAD

**Issue by gabrielbauman** [twineworks/tweakflow#3]
Thanks for this library; a great little language and the implementation is pretty tight. I'm enjoying looking at the internals.

For the templating use case, it would be nice to have a more compact syntax, such as allowing the evaluation of functions within a string:

`"First name: #{strings.upper_case(first_name)}"`

or, perhaps a filter syntax

`"First name: #{first_name|strings.upper_case|...}"`

I haven't looked at feasibility, just wanted to see what the thinking is.

**Comment by slawo-ch** [comment]
Thanks Gabriel,

I agree with the templating use case. It's been on my list to try and knock out for a while. The difficulty lies in the Lexer->Parser approach. So far I have not found a simple way to convince a single lexer and parser to recursively descend and re-parse the contents of an interpolation. I'll have a look at it again before version 1.0 comes around. I'd very much like that.

As for the filter syntax, that's probably more feasible to do, but it would be sort of a crutch ontop of the existing crutch of allowing references only. I'd rather go all in and allow sub-expressions.

String concatenation is `..` right now for no other reason than to be distinct from creating sums. If you have numbers that go next to each other in a concatenation `x+y+" foo"` would create a sum first, then concat it with "foo".  I agree that an overloaded `+` is probably not an issue in practice, but keeping things separate has its merits too, I think.

**Comment by gabrielbauman** [comment]
I had just discovered the `..` operator when I saw your response. I would lean toward **not** overloading `+` if it were up to me, but then I'm an old guy who liked `:=` to differentiate assignment from equality in Object Pascal ;). I like your design choices overall.

Thanks for your time. I'll play with this some more, and if I do anything interesting I'll send a PR. Sub-expressions would be nifty.
